core/Types.h

```diff
@@ -160,6 +160,9 @@ class Types final {
     static TypePtr widen(Context ctx, const TypePtr &type);
     static std::optional<int> getProcArity(const AppliedType &type);
 
+    /** Unwrap SelfTypeParam instances that belong to the given owner, to a bare LambdaParam */
+    static TypePtr unwrapSkolemVariables(Context ctx, const TypePtr &ty);
```

Comment:
the word `Skolem` has waay to many meanings and this is the first time it's used.
WDYT about `unwrapSelfTypeParams`?

Reply:
Good point, I'll make that change. Thanks!